src/gt4py/next/program_processors/runners/dace_fieldview/transformations/k_blocking.py

```diff
@@ -0,0 +1,437 @@
+        if scope[map_entry] is not None:
+            return False
+        if block_var not in map_entry.map.params:
+            return False
+        if map_entry.map.schedule == dace.dtypes.ScheduleType.Sequential:
+            return False
+        if map_range[map_params.index(block_var)][2] != 1:
+            return False
+        if self.partition_map_output(map_entry, block_var, graph, sdfg) is None:
+            return False
+
+        return True
+
+    def apply(
+        self,
+        graph: Union[SDFGState, SDFG],
+        sdfg: SDFG,
+    ) -> None:
+        """Creates a blocking map.
+
+        Performs the operation described in the doc string.
+        """
+        outer_entry: dace_nodes.MapEntry = self.map_entry
+        outer_exit: dace_nodes.MapExit = graph.exit_node(outer_entry)
+        outer_map: dace_nodes.Map = outer_entry.map
+        map_range: dace_subsets.Range = outer_entry.map.range
+        map_params: list[str] = outer_entry.map.params
+
+        # This is the name of the iterator we coarsen
+        block_var: str = self.block_dim
+        block_idx = map_params.index(block_var)
+
+        # This is the name of the iterator that we use in the outer map for the
+        #  blocked dimension
+        coarse_block_var = "__coarse_" + block_var
+
+        # Now compute the partitions of the nodes.
+        independent_nodes, dependent_nodes = self.partition_map_output(  # type: ignore[misc]  # Guaranteed to be not `None`.
+            outer_entry, block_var, graph, sdfg
+        )
+
+        # Generate the sequential inner map
+        rng_start = map_range[block_idx][0]
+        rng_stop = map_range[block_idx][1]
+        inner_label = f"inner_{outer_map.label}"
+        inner_range = {
+            block_var: dace_subsets.Range.from_string(
+                f"({coarse_block_var} * {self.blocking_size} + {rng_start}):min(({rng_start} + {coarse_block_var} + 1) * {self.blocking_size}, {rng_stop} + 1)"
+            )
+        }
+        inner_entry, inner_exit = graph.add_map(
+            name=inner_label,
+            ndrange=inner_range,
+            schedule=dace.dtypes.ScheduleType.Sequential,
+        )
+
+        # TODO(phimuell): Investigate if we want to prevent unrolling here
+
+        # Now we modify the properties of the outer map.
+        coarse_block_range = dace_subsets.Range.from_string(
+            f"0:int_ceil(({rng_stop} + 1) - {rng_start}, {self.blocking_size})"
+        ).ranges[0]
+        outer_map.params[block_idx] = coarse_block_var
+        outer_map.range[block_idx] = coarse_block_range
+        outer_map.label = f"{outer_map.label}_blocked"
+
+        # Contains the independent nodes that are already relocated.
+        relocated_nodes: set[dace_nodes.Node] = set()
+
+        # Now we iterate over all the output edges of the outer map and rewire them.
+        #  Note that this only handles the entry of the Map.
+        for out_edge in list(graph.out_edges(outer_entry)):
```

Comment:
Is a list instantiation really needed here? 
```suggestion
        for out_edge in graph.out_edges(outer_entry):
```

Reply:
I darkly remember that it was, because you are gonna manipulate that very set of edges.
However, I looked at the DaCe source and it seems that they do that already, so it should be technically safe to do do. But it seems that it would not be save if we use the networkx layer directly.

Reply:
I removed them where it is safe to do.
I also asked them and they say that it was not specifically designed in that way. However, they do not expect to change it.